doc/specs/#1043 - Set the initial position of the Terminal/spec.md

```diff
@@ -0,0 +1,104 @@
+---
+author: Kaiyu Wang KaiyuWang16/kawa@microsoft.com
+created on: 2019-09-03
+last updated: 2019-09-03
+issue id: #1043
+---
+
+# Set the initial position for terminal
+
+## Abstract
+
+This spec is for task #1043 �Be able to set an initial position for the terminal�. It goes over the details of a new feature that allows users to set the initial position and size of the terminal. Expected behavior and design of this feature is included. Besides, future possible follow-up works are also addressed.
+
+## Inspiration
+
+The idea is to allow users to set the initial position of the Terminal when they launch it, prevent the Terminal from appearing on unexpected position (e.g. outside of the screen bounds). We are also going to let users choose to maximize the window when they launch it.
+
+## Solution Design
+
+For now, the Terminal window is put on a default initial position. The program uses CW_USEDEFAULT in the screen coordinates for top-left corner. We have two different types of window � client window and non-client window. However, code path for window creation (WM_CREATE message is shared by the two types of windows) are almost the same for the two types of windows, except that there are some differences in calculation of the width and height of the window.
+
+Three new properties should be added in the json settings file:
+
+**initialX**: int. This set the initial horizontal position of the top-left corner of the window. This property is optional. If not provided, system default position will be used. 
+
+**initialY**: int. This set the initial vertical position of the top-left corner of the window. This property is optional. If not provided, system default position will be used.
+
+**launchMode**: string. Determine the launch mode. There are two modes for now
+                         1. maximize: the window will be maximized when launch.
+                         2. default: the window will be initialized with system default size. 
+
+The steps of this process:
+
+1. Set the top-left origin, width and height to CW_USEDEFAULT.
+2. Get the dpi of the nearest monitor; Load settings.
+3. From settings, find the user-defined initial position and launch mode. 
+4. If the user sets custom initial position, calculate the new position considering the current dpi and monitor. If not, use system default value.
+5. If the user set launch mode as "maximize", calculate the new height and width. If the user choose "default", use system default size.

```

Comment:
Will the position be relative to whatever monitor the system selected? The user will not be able to control the monitor in use, correct?

Reply:
Yes, I think so. In settings->Display, we can see how the monitors are arranged in the system: 

![image](https://user-images.githubusercontent.com/11893272/66610907-d17d6a80-eb71-11e9-9c89-73dbc267969d.png)

And the coordinates depend on this figure. It is also related where the main display is. The top-left corner of the main display is (0, 0). We can rearrange the layout of the monitors or change the main display in settings.